agot-bg-game-server/src/common/ingame-game-state/game-data-structure/westeros-card/WinterIsComingWesterosCardType.ts

```diff
@@ -12,16 +12,17 @@ export default class WinterIsComingWesterosCardType extends WesterosCardType {
             westerosDeckI: currentDeckI
         });
 
-        const deck = westerosGameState.game.westerosDecks[currentDeckI];
+        let deck = westerosGameState.game.westerosDecks[currentDeckI];
+        const dontShuffle = Math.max(0,westerosGameState.game.revealedWesterosCards - 1);
 
         // Reset discarded state
         deck.forEach(card => { card.discarded = false; });
 
         // Shuffle the deck
-        shuffle(deck);
+        deck = deck.slice(0, dontShuffle).concat(shuffle(deck.slice(dontShuffle)));
```

Comment:
The WiC situation is more complex than I initially thought. We can't do it that way because the current WiC wouldn't be taken into account when shuffling the deck but according to the rulebook it has to:

> The "Winter is Coming" Westeros Card provides the
only element of uncertainty in this variant. Resolve this
card as you would in the regular game by placing the
card back in its respective deck, shuffling the deck, and
drawing a new Card which is immediately resolved as if
it were the top-most card of the column.

So in fact only card 2 and 3 are fixed and shouldn't be shuffled.

Reply:
So the reason why this code works is very simple. Card drawing works like that

Deck: [a,b,c,d,e]

1) Get the first card of the deck and shifts it. 
Card [a], Deck [b,c,d,e]

2) Adds discarded tag and pushes the card to the end of the array
Deck: [b,c,d,e,a]

3) Execute effects

So by the time WiC is getting resolved, cards **2 and 3** are actually cards **1 and 2**. First, I wrote considering cards 2 and 3 but after looking at _src/common/ingame-game-state/westeros-game-state/WesterosGameState_ I realized that this was the proper way to do it

Reply:
Thx for explaining our own code.. 😄 I got it... I still were at the thought WiC is the first card at that point but you are right. It has been already drawn and shifted to the bottom. Maybe we should do this shift after a card is resolved to be more like it would be done in a real world game. But what is done is done and we keep it as it is of course.